test/Dockerfile

```diff
@@ -5,6 +5,13 @@ RUN apk add openssh-server openssh file curl
 RUN sed -E "s/^#*(PermitRootLogin ).+$/\1yes/" /etc/ssh/sshd_config -i
 RUN sed -E "s/^#*(AllowTcpForwarding ).+$/\1yes/" /etc/ssh/sshd_config -i
 RUN ssh-keygen -A
-RUN echo "root:$root_password"|chpasswd
+# Install [gost](https://github.com/ginuerzh/gost) for testing proxy support.
+# Run this before changing root password to preseve the docker layer caching as
+# much as possible across tests.
+RUN wget -O /tmp/gost.gz https://github.com/ginuerzh/gost/releases/download/v2.11.5/gost-linux-amd64-2.11.5.gz \
+    && gunzip -c -k /tmp/gost.gz > /usr/local/bin/gost \
+    && chmod +x /usr/local/bin/gost \
+    && rm -f /tmp/gost.gz
 ENTRYPOINT ["/usr/sbin/sshd","-D","-E","/tmp/sshd.log"]
-EXPOSE 22/tcp
\ No newline at end of file
+EXPOSE 22/tcp
+RUN echo "root:$root_password"|chpasswd
```

Comment:
This line is moved to the bottom of the Dockerfile so all the previous layers (including "EXPOSE" and "ENTRYPOINT") could be cached even when `root_password` arg is changed.